pkg/rke2/spw.go

```diff
@@ -42,57 +26,55 @@ func checkStaticManifests(dataDir string) cmds.StartupHook {
 	return func(ctx context.Context, wg *sync.WaitGroup, args cmds.StartupHookArgs) error {
 		go func() {
 			defer wg.Done()
+			if err := wait.PollImmediate(20*time.Second, 30*time.Minute, func() (bool, error) {
 
-			var conn *grpc.ClientConn
-			if err := wait.ExponentialBackoff(criBackoff, func() (done bool, err error) {
-				conn, err = containerdk3s.CriConnection(ctx, containerdSock)
+				conn, err := containerdk3s.CriConnection(ctx, containerdSock)
 				if err != nil {
 					logrus.Infof("Waiting for cri connection: %v", err)
 					return false, nil
 				}
-				return true, nil
-			}); err != nil {
-				logrus.Fatalf("failed to setup cri connection: %v", err)
-			}
-			cRuntime := runtimeapi.NewRuntimeServiceClient(conn)
-			defer conn.Close()
+				cRuntime := runtimeapi.NewRuntimeServiceClient(conn)
+				defer conn.Close()
 
-			manifestDir := podManifestsDir(dataDir)
+				manifestDir := podManifestsDir(dataDir)
 
-			for _, pod := range []string{"etcd", "kube-apiserver"} {
-				manifestFile := filepath.Join(manifestDir, pod+".yaml")
-				if f, err := os.Open(manifestFile); err == nil {
-					podManifest := v1.Pod{}
-					decoder := yaml.NewYAMLToJSONDecoder(f)
-					err = decoder.Decode(&podManifest)
-					if err != nil {
-						logrus.Fatalf("Failed to decode %s manifest: %v", pod, err)
-					}
-					podFilter := &runtimeapi.ContainerFilter{
-						LabelSelector: map[string]string{
-							"io.kubernetes.container.name": pod,
-						},
-					}
-					if err := wait.ExponentialBackoff(podCheckBackoff, func() (done bool, err error) {
+				for _, pod := range []string{"etcd", "kube-apiserver"} {
+					manifestFile := filepath.Join(manifestDir, pod+".yaml")
+					if f, err := os.Open(manifestFile); err == nil {
+						podManifest := v1.Pod{}
+						decoder := yaml.NewYAMLToJSONDecoder(f)
+						err = decoder.Decode(&podManifest)
+						if err != nil {
+							logrus.Fatalf("Failed to decode %s manifest: %v", pod, err)
+						}
+						podFilter := &runtimeapi.ContainerFilter{
+							LabelSelector: map[string]string{
+								"io.kubernetes.container.name": pod,
+							},
+						}
 						resp, err := cRuntime.ListContainers(ctx, &runtimeapi.ListContainersRequest{Filter: podFilter})
 						if err != nil {
 							return false, err
 						}
+						found := false
 						for _, c := range resp.Containers {
 							if c.Labels["io.kubernetes.pod.uid"] == string(podManifest.UID) {
 								logrus.Infof("Latest %s manifest deployed", pod)
-								return true, nil
+								found = true
 							}
 						}
-						logrus.Infof("Waiting for %s manifest", pod)
-						return false, nil
-					}); err != nil {
-						logrus.Fatalf("Failed to wait for latest %s manifest to be deployed: %v", pod, err)
+						if !found {
```

Comment:
If we filter containers on `io.kubernetes.pod.uid` then we can just do:
```suggestion
						if len(resp.Containers) < 1 {
```